be/src/exec/vectorized/hdfs_scanner_orc.cpp

```diff
@@ -499,15 +500,18 @@ Status HdfsOrcScanner::do_get_next(RuntimeState* runtime_state, ChunkPtr* chunk)
         orc::RowReader::ReadPosition position;
         size_t read_num_values = 0;
         bool has_used_dict_filter = false;
+        ColumnPtr row_delete_filter = BooleanColumn::create();
         {
             SCOPED_RAW_TIMER(&_stats.column_read_ns);
             RETURN_IF_ERROR(_orc_reader->read_next(&position));
+            row_delete_filter = _orc_reader->get_row_delete_filter(_deleted_position_set);
             // read num values is how many rows actually read before doing dict filtering.
             read_num_values = position.num_values;
             RETURN_IF_ERROR(_orc_reader->apply_dict_filter_eval_cache(_orc_row_reader_filter->_dict_filter_eval_cache,
                                                                       &_dict_filter));
             if (_orc_reader->get_cvb_size() != read_num_values) {
                 has_used_dict_filter = true;
+                row_delete_filter->filter(_dict_filter);
```

Comment:
Is it duplicate with LOC:579?

Reply:
row_delete_filter should be considered as an active slot but not in reader's batch (virtual __POSITION).
in `apply_dict_filter_eval_cache` we may filter active columns in batch. If that is done, row_delete_filter should be filtered too or the POSITION is incorrect